File: app/src/main/java/com/ytempest/wanandroid/base/fragment/MvpFragment.java
package com.ytempest.wanandroid.base.fragment;

import android.content.Context;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import android.view.View;

import com.ytempest.tool.util.ToastUtils;
import com.ytempest.wanandroid.base.presenter.IPresenter;
import com.ytempest.wanandroid.base.view.IView;
import com.ytempest.wanandroid.dialog.LoadingDialog;

import javax.inject.Inject;

import dagger.android.support.AndroidSupportInjection;

/**
 * @author heqidu
 * @since 2020/6/30
 */
public abstract class MvpFragment<Presenter extends IPresenter> extends AbstractFragment implements IView {

    @Inject
    protected Presenter mPresenter;

    @Override
    public void onAttach(Context context) {
        AndroidSupportInjection.inject(this);
        super.onAttach(context);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (mPresenter != null) {
            mPresenter.attachView(this);
        }
    }

    @Override
    public void onDestroyView() {
        if (mPresenter != null) {
            mPresenter.detach();
        }
        super.onDestroyView();
    }


    /*View*/

    @Override
    public void showToast(String msg) {
        ToastUtils.show(getContext(), msg);
    }

    @Override
    public void showToast(@StringRes int textId) {
        ToastUtils.show(getContext(), textId);
    }

    private LoadingDialog mLoadingDialog;

    @Override
    public void showLoading() {
        if (mLoadingDialog == null) {
            mLoadingDialog = new LoadingDialog(getContext());
            mLoadingDialog.setAutoDismiss(getLifecycle());
        }
        mLoadingDialog.show();
    }

    @Override
    public void stopLoading() {
        if (mLoadingDialog != null) {
            mLoadingDialog.dismiss();
        }
    }
}

File: app/src/main/java/com/ytempest/wanandroid/base/fragment/MvpFragment.kt
package com.ytempest.wanandroid.base.fragment

import android.content.Context
import android.os.Bundle
import android.view.View
import androidx.viewbinding.ViewBinding
import com.ytempest.wanandroid.base.presenter.IPresenter
import com.ytempest.wanandroid.base.view.IView
import dagger.android.support.AndroidSupportInjection
import javax.inject.Inject

/**
 * @author heqidu
 * @since 21-2-7
 */
abstract class MvpFragment<Presenter : IPresenter, VB : ViewBinding> : AbstractFragment<VB>(), IView {

    @Inject
    protected lateinit var mPresenter: Presenter

    override fun onAttach(context: Context) {
        AndroidSupportInjection.inject(this)
        super.onAttach(context)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mPresenter.attachView(this)
    }

    override fun onDestroyView() {
        mPresenter.detach()
        super.onDestroyView()
    }
}